spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfigurationTests.java

```diff
@@ -332,11 +336,30 @@ public void testRabbitTemplateBackOff() {
 	@Test
 	public void testRabbitMessagingTemplateBackOff() {
 		this.contextRunner.withUserConfiguration(TestConfiguration4.class)
+				.withPropertyValues("spring.rabbitmq.listener.simple.retry.enabled=true",
```

Comment:
I am not sure I understand the purpose of that test. We're creating a messaging template to show that it's taken into account. Why is retry processing added there?

Reply:
I am testing that the customizer injected a listener into the listener container's retry template; in order to test that, we need to enable retry.

Reply:
I got that. But why are you testing that as part of `testRabbitMessagingTemplateBackOff` ?